File: Data Structures & Algorithms/implement-stack-using-queues/submission-0.java
class MyStack {
    public Queue<Integer> q; 
    public Queue<Integer> q1;

    public MyStack() {
        q = new LinkedList<>();
        q1 = new LinkedList<>();
    }
    
    public void push(int x) {
        q.offer(x); 
    }
    
    public int pop() {
        while (q.size() > 1) {
            q1.offer(q.poll());
        }
        int res = q.poll();
        Queue<Integer> temp = q;
        q = q1;
        q1 = temp; 
        return res;       
    }
    
    public int top() {
        while (q.size() > 1) {
            q1.offer(q.poll());
        }
        int res = q.poll();
        q1.offer(res);
        Queue<Integer> temp = q;
        q = q1;
        q1 = temp;
        return res;  
    }
    
    public boolean empty() {
        return q.size() == 0;        
    }
}

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack obj = new MyStack();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.top();
 * boolean param_4 = obj.empty();
 */

File: Data Structures & Algorithms/implement-queue-using-stacks/submission-0.py
class MyQueue:

    def __init__(self):
        self.stack = []        

    def push(self, x: int) -> None:
        self.stack.append(x)        

    def pop(self) -> int:
        res = self.stack[0]
        #for i in range(len(self.stack)-1):
        #   self.stack[i] = self.stack[i+1]
        self.stack.pop(0)
        return res
        

    def peek(self) -> int:
        return self.stack[0]
        

    def empty(self) -> bool:
        if self.stack:
            return False
        return True
        


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()